### DISCUSSION TITLE
Order by random()?

### DISCUSSION THREAD

**Discussion by mofax** [supabase/supabase#14609] [Questions]
Postgres allows you to fetch a random subset of rows either by

```sql
select * from table_name order by random() limit 5;
```
or
```sql
SELECT *
FROM table_name
TABLESAMPLE BERNOULLI (10);
```

How can I achieve either using the supabase client?

**Comment by mofax** [comment]
In the interim, to keep moving, I have created a function
```sql
CREATE OR REPLACE function fetch_random_words()
returns setof words
language sql
as $$ 
   SELECT * FROM words TABLESAMPLE BERNOULLI (5)
$$;
```

**Comment by TomSinclair** [comment]
Is there any update on if this is likely to be added?

**Reply by GaryAustin1** [reply]
https://github.com/PostgREST/postgrest-docs/issues/671
Seems to say no.
It requires a full table scan according to one of the PostgREST devs and they don't want to make that easy to do for performance reasons.
